### DISCUSSION TITLE
Error while running searchable deploy | f"Could not find a connection with guid '{connection_guid}

### DISCUSSION THREAD

**Discussion by bsingh15** [thoughtspot/cs_tools#132] [General] [answered]
Hi All,

I have been trying to deploy the searchable deploy app in my ThoughtSpot error and seems like it is not able to find a connection for the guid I provided.

Here is the command I ran -

cs_tools tools searchable deploy --config tegna_prod --connection-guid '467e69bf-49f3-4d94-8cc0-9cd39d9e2eb9' --database 'dev_ope_raw_data' --schema 'cs_tools'

Please find attached two files -
1. A screenshot of how I got my connection GUID
2. A screenshot of the stack trace

I also just ran cs_tools logs report right after I got this error at around 3:30 EST on 5/7/24. Here is the 3 line output from this command -

PS C:\Users\bsingh> type 2024-05-07T19_30_59.log
[DEBUG - 2024-05-07T15:31:20-0400] [cs_tools.cli._analytics - _analytics.maybe_send_analytics_data 121] /analytics/runtimes :: <Response [201 Created]>
[DEBUG - 2024-05-07T15:31:24-0400] [cs_tools.cli._analytics - _analytics.maybe_send_analytics_data 129] /analytics/commands :: <Response [201 Created]>
[DEBUG - 2024-05-07T15:31:24-0400] [cs_tools.cli._analytics - _analytics.maybe_send_analytics_data 137] Sent analytics to CS Tools!
PS C:\Users\bsingh>

Please take a look and let me know what we can do to make this work.

Thank you,
Bhupinder
[ts_searchable_deploy_stack_trace.docx](https://github.com/thoughtspot/cs_tools/files/15240052/ts_searchable_deploy_stack_trace.docx)
[How_where_to_get_connection_guid_from.docx](https://github.com/thoughtspot/cs_tools/files/15240053/How_where_to_get_connection_guid_from.docx)

**Comment by boonhapus** [comment]
Thanks @bsingh15 , I'm traveling today but I will look into it and get back to you.

**Reply by boonhapus** [reply]
@bsingh15 this should be fixed in the latest version. You can fetch it with..

```
powershell -c "IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py | python - --install"
```

**Reply by bsingh15** [reply]
Hi @boonhapus, I downloaded and installed latest version of the cs_tools -
powershell -c "IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py | python - --install"
         | Building wheels for collected packages: cs_tools
         | Building wheel for cs_tools (pyproject.toml): started
         | Building wheel for cs_tools (pyproject.toml): finished with status 'done'
         | Created wheel for cs_tools: filename=cs_tools-1.5.5-py3-none-any.whl size=3515142 sha256=d2df5f9f39735a22511ed1227ca823f7bec3a78956745c8c0cc8f84c45a4038c
         | Stored in directory: C:\Users\bsingh\AppData\Local\Temp\pip-ephem-wheel-cache-bu0riys0\wheels\4a\f8\95\237f768eef74e0271245121e871245a58b28185c83f8f2bf92
         | Successfully built cs_tools
         | Installing collected packages: cs_tools
         | Attempting uninstall: cs_tools
         | Found existing installation: cs_tools 1.5.3
         | Uninstalling cs_tools-1.5.3:
         | Successfully uninstalled cs_tools-1.5.3
         | Successfully installed cs_tools-1.5.5
         | Done! Thank you for trying CS Tools.
         | You're almost there! Please restart your shell and then execute the command below.
         |
         | cs_tools --version
         |

Now, I am getting table not found errors. Please find attached two pages from the error output.

![searchable_deploy_tables_not_found_errors_page2](https://github.com/thoughtspot/cs_tools/assets/105447834/f878ce17-63ec-4cb8-8dd0-a9e78ce75438)
![searchable_deploy_tables_not_found_errors_page1](https://github.com/thoughtspot/cs_tools/assets/105447834/d57467e1-a694-4bd6-aa3f-581fa799517e)

**Reply by boonhapus** [reply]
@bsingh15 What Coinnection type are you using? The TML is pointing to

```
database = dev_ope_raw_data
schema = cs_tools
```

It looks like the table names are not getting lowercased, which should be happening if you're on any other dialect than Snowflake, but that doesn't appear to be working. 😓

**Reply by boonhapus** [reply]
Actually -- have you run `searchable metadata` and `searchable bi-server` once yet?

**Reply by bsingh15** [reply]
The connection is snowflake. Snowflake does not care if a name is lower or upper unless its enclosed in double quotes.

I had run searchable metadata & bi-server twice. Once before I upgraded to V1.5.5 and once after just to rule out running it again. :). Also, for your reference, here is the complete snowflake toml file. (I stubbed out sensitive info like account, secret)

PS C:\Users\bsingh\cs_tools> type snowflake_cs_tools.toml
[configuration]
account_name = '**'
username = '**'
secret = '**'
warehouse = 'tegna_dev_wh'
role = 'tegna_dev'
database = 'dev_ope_raw_data'
schema = 'cs_tools'
authentication = 'basic'
load_strategy = 'truncate'

PS C:\Users\bsingh\cs_tools>

Question: Is there a way to test the toml file itself? Like it can connect to snowflake and see the tables without having to run the whole deploy command?

**Reply by boonhapus** [reply, marked as answer]
> Question: Is there a way to test the toml file itself? Like it can connect to snowflake and see the tables without having to run the whole deploy command?

@bsingh15 I don't document this well. The Syncer TOML is for data loads only. The deploy command uses ThoughtSpot APIs and is only there to put the TML from the repo, into your ThoughtSpot system -- nothing more. It can be run once and then never again.

Snowflake doesn't care about case sensitivity unless you quote, but I would not be surprised if ThoughtSpot's tml code actually does quote object names. This is a pattern that is widely used which may cause issues. I would recommend trying your deploy command use the exact casing that exists in Snowflake for the database and schema.

The Syncer from the CS Tools project uses **[snowflake-sqlalchemy](https://github.com/snowflakedb/snowflake-sqlalchemy)**, which I know for a fact does not quote object names _unless_ you enforce it, which is why your data loads work fine.

**Reply by bsingh15** [reply]
That was it. Easy/Breezy. I had to change the name of the database and the schema to be all caps in the deploy command and it worked!

cs_tools tools searchable deploy --config tegna_prod --connection-guid '467e69bf-49f3-4d94-8cc0-9cd39d9e2eb9' --database 'DEV_OPE_RAW_DATA' --schema 'CS_TOOLS'

ThoughtSpot does put object names in double quotes when it calls database. I know since I sometimes use ThoughtSpot generated SQLs to debug my logic.

So, in this case since we are passing the database name and the schema name as a string from the command line, it takes them as verbatim and encloses in double quotes and this is how snowflake sees them and can't find the database or the schema!

**Reply by bsingh15** [reply]
Whereas I think when you process the toml file you are converting to all caps before using the details there. Otherwise, that would not have worked either since I have a tendency to use lower case most of the times!
